pkg/dynamic/lib.go

```diff
@@ -0,0 +1,125 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamic
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// gvrFromUnstructured takes a unstructured object of CRD type and finds GVR from its spec
+func gvrFromUnstructured(u *unstructured.Unstructured) (gvr schema.GroupVersionResource, err error) {
```

Comment:
I'm still pretty sure that there must be a library call for doing this (I remember very darkly something like `GetGvrUnsafe()` or something like this. but  not sure).

I suggest to merge this in, but if we find out about that library replace it here.

Reply:
my bad, its about parsing the spec's versions etc. which is kind of quite unique.